### PR TITLE
Add ble temp example

### PR DESCRIPTION
**Pull request by glennrub**

### CODE REVIEW

File: nrf5/hal/hal_temp.c [hunk 1]
```diff
@@ -25,18 +25,36 @@
  */
 
 #include <stdio.h>
+#include <stdint.h>
 #include "mphalport.h"
 #include "hal_temp.h"
 
+#if BLUETOOTH_SD
```
Comment (tralamazza) [inline nrf5/hal/hal_temp.c:32]: do we want BLE stuff inside the peripheral hal layer?

Comment (glennrub) [reply nrf5/hal/hal_temp.c:32]: it has to.. as it must use the softdevice version of temp access. It cannot go directly on the peripheral. The other solution is to make a HAL-hal.

Comment (tralamazza) [reply nrf5/hal/hal_temp.c:32]: or have that `sd_temp_get()` in `machine_temp_read()` temp.c

Comment (glennrub) [reply nrf5/hal/hal_temp.c:32]: You are right, that could have worked. Until now i have not interfaced the sd_* other than through the ble_drv driver, trying to abstract it. Similar #ifdef is also present in hal_irq for nvic access.

To add in, other peripherals that would need similar abstraction to detect whether bluetooth stack is enabled or not would be:
- PPI 
- Flash (non-volatile memory)
- RNG
- System off
- System reset
- Reset reason get/clear
- RAMON/OFF
- GPREGRET (register retention)
- HFCLK 
- ECB
- Radio (in parallel with BLE)

Most of these does not have its own module, but might be used by drivers as i can see it. The idea is to provide an reusable interface to the HAL whether or not bluetooth is enabled or not.

Comment (tralamazza) [reply nrf5/hal/hal_temp.c:32]: it all depends if it's OK for us to have SD_ functions inside the HAL layer. My vote is for a SD agnostic pure HAL layer

Comment (tralamazza) [reply nrf5/hal/hal_temp.c:32]: OK so we will have SD functions in the HAL to simplify

**Pull request merged**